src/librustc/ich/impls_hir.rs

```diff
@@ -47,33 +47,14 @@ impl<'ctx> rustc_hir::HashStableContext for StableHashingContext<'ctx> {
     // want to pick up on a reference changing its target, so we hash the NodeIds
     // in "DefPath Mode".
```

Comment:
I think it makes sense to move this comment back to the `HashStable` impls for the `ItemId` types.